tests/e2e_tests/multistep/test_incentive.py

```diff
@@ -0,0 +1,287 @@
+                    break
+                f.write(line.decode())
+                f.flush()
+
+    # Create tasks to read stdout and stderr concurrently
+    asyncio.create_task(validator_write_output(validator_process.stdout))
+    asyncio.create_task(validator_write_output(validator_process.stderr))
+
+    await asyncio.sleep(
+        5
+    )  # wait for 5 seconds for the metagraph and subtensor to refresh with latest data
+
+    # register validator with root network
+    alice_exec_command(
+        RootRegisterCommand,
+        [
+            "root",
+            "register",
+            "--netuid",
+            "1",
+            "--wallet.name",
+            "default",
+            "--wallet.hotkey",
+            "default",
+            "--subtensor.chain_endpoint",
+            "ws://localhost:9945",
+        ],
+    )
+
+    alice_exec_command(
+        RootSetBoostCommand,
+        [
+            "root",
+            "boost",
+            "--netuid",
+            "1",
+            "--increase",
+            "100",
+            "--wallet.name",
+            "default",
+            "--wallet.hotkey",
+            "default",
+            "--subtensor.chain_endpoint",
+            "ws://localhost:9945",
+        ],
+    )
+
+    # get latest metagraph
+    metagraph = bittensor.metagraph(netuid=1, network="ws://localhost:9945")
+
+    # get current emissions
+    bob_neuron = metagraph.neurons[1]
+    assert bob_neuron.incentive == 0
+    assert bob_neuron.consensus == 0
+    assert bob_neuron.rank == 0
+    assert bob_neuron.trust == 0
+
+    alice_neuron = metagraph.neurons[0]
+    assert alice_neuron.validator_permit is False
+    assert alice_neuron.dividends == 0
+    assert alice_neuron.stake.tao == 9999.999999
+    assert alice_neuron.validator_trust == 0
+
+    # wait until 360 blocks pass (subnet tempo)
+    wait_epoch(360, subtensor)
+
+    # for some reason the weights do not get set through the template. Set weight manually.
+    alice_wallet = bittensor.wallet()
+    alice_wallet._hotkey = alice_keypair
+    subtensor._do_set_weights(
+        wallet=alice_wallet,
+        uids=[1],
+        vals=[65535],
+        netuid=1,
+        version_key=0,
+        wait_for_inclusion=True,
+        wait_for_finalization=True,
+    )
+
+    # wait epoch until weight go into effect
+    wait_epoch(360, subtensor)
+
+    # refresh metagraph
+    metagraph = bittensor.metagraph(netuid=1, network="ws://localhost:9945")
+
+    # get current emissions and validate that Alice has gotten tao
+    bob_neuron = metagraph.neurons[1]
+    assert bob_neuron.incentive == 1
+    assert bob_neuron.consensus == 1
+    assert bob_neuron.rank == 1
+    assert bob_neuron.trust == 1
+
+    alice_neuron = metagraph.neurons[0]
+    assert alice_neuron.validator_permit is True
+    assert alice_neuron.dividends == 1
+    assert alice_neuron.stake.tao == 9999.999999
+    assert alice_neuron.validator_trust == 1
+
+
+def wait_epoch(interval, subtensor):
```

Comment:
nitpick: Usually functions-helpers are placed before the functions that call it. It's just easier to navigate.